tests/forms_tests/widget_tests/test_selectdatewidget.py

```diff
@@ -314,6 +314,68 @@ class GetRequiredDate(Form):
         self.assertFalse(GetNotRequiredDate().fields['mydate'].widget.is_required)
         self.assertTrue(GetRequiredDate().fields['mydate'].widget.is_required)
 
+    def test_selectdate_placeholder(self):
+        class TestForm(Form):
+            mydate = DateField(widget=SelectDateWidget(years=('2018', '2019')), required=True)
+
+        self.check_html(TestForm().fields['mydate'].widget, 'mydate', '', html=(
+            """
+            <select name="mydate_month" id="id_mydate_month" placeholder="Month">
+                <option value="1">January</option>
+                <option value="2">February</option>
+                <option value="3">March</option>
+                <option value="4">April</option>
+                <option value="5">May</option>
+                <option value="6">June</option>
+                <option value="7">July</option>
+                <option value="8">August</option>
+                <option value="9">September</option>
+                <option value="10">October</option>
+                <option value="11">November</option>
+                <option value="12">December</option>
+            </select>
+
```

Comment:
I think you can remove the blank lines as the helpers for checking HTML should ignore whitespace between tags.

Reply:
No, It doesn't. I added a test to cover the `required=False` case.